src/main/scala/scorex/api/http/BlocksApiRoute.scala

```diff
@@ -104,19 +104,25 @@ case class BlocksApiRoute(settings: RestAPISettings, checkpointsSettings: Checkp
   @ApiImplicitParams(Array(
     new ApiImplicitParam(name = "height", value = "Block height", required = true, dataType = "integer", paramType = "path")
   ))
-  def at: Route = (path("at" / IntNumber) & get)(at(_, includeTransactions = true))
+  def at: Route = (path("at" / IntNumber) & get) (at(_, includeTransactions = true))
 
   @Path("/headers/at/{height}")
   @ApiOperation(value = "At(Block header only)", notes = "Get block at specified height without transactions payload", httpMethod = "GET")
   @ApiImplicitParams(Array(
     new ApiImplicitParam(name = "height", value = "Block height", required = true, dataType = "integer", paramType = "path")
   ))
-  def atHeaderOnly: Route = (path("headers" / "at" / IntNumber) & get)(at(_, includeTransactions = false))
+  def atHeaderOnly: Route = (path("headers" / "at" / IntNumber) & get) (at(_, includeTransactions = false))
 
-  private def at(height: Int, includeTransactions: Boolean): StandardRoute =
-    history.blockAt(height).map(_.json(includeTransactions)) match {
-    case Some(json) => complete(json + ("height" -> JsNumber(height)))
-    case None => complete(Json.obj("status" -> "error", "details" -> "No block for this height"))
+  private def at(height: Int, includeTransactions: Boolean): StandardRoute = {
+    (if (includeTransactions) {
+      history.blockAt(height).map(_.json)
+    }
+    else {
+      history.blockHeaderAndSizeAt(height).map(x => x._1.json(x._2))
```

Comment:
next time try to use pattern-matching where possible
```
map { case((header,size)) => ...
```

Reply:
yes, my liege